lib/IDE/IDETypeChecking.cpp

```diff
@@ -327,7 +327,7 @@ struct SynthesizedExtensionAnalyzer::Implementation {
         case RequirementKind::Superclass:
           // FIXME: This could be more accurate; check
           // conformance instead of subtyping
-          if (!canPossiblyConvertTo(First, Second, *DC))
+          if (!isConvertibleTo(First, Second, /*openArchetypes=*/true, *DC))
```

Comment:
for a Superclass constraint, you want to check isBindableToSuperclassOf() (that's like the "open archetypes" version), followed by isExactSuperclassOf().

For conformance, we'll need to cook something up. Basically a type parameter "can potentially" conform to anything; for a concrete type you call back to conformsToProtocol().